mypy/checker.py

```diff
@@ -2674,6 +2684,52 @@ def check_member_assignment(self, instance_type: Type, attribute_type: Type,
         infer = is_subtype(rvalue_type, get_type) and is_subtype(get_type, set_type)
         return rvalue_type if infer else set_type, get_type, infer
 
+    def check_parent_member_assignment(self, lvalue_type: Type,
+                                       lvalue_attribute_type: Type, rvalue_type: Type,
```

Comment:
The naming here is also confusing. `lvalue_type` is actually `lvalue_base_type`, while `lvalue_attribute_type` is just `lvalue_type`.